stackdriver_exporter.go

```diff
@@ -210,28 +214,46 @@ func newHandler(projectIDs []string, metricPrefixes []string, metricExtraFilters
 		metricsExtraFilters: metricExtraFilters,
 		additionalGatherer:  additionalGatherer,
 		m:                   m,
+		collectors:          map[string]*collectors.MonitoringCollector{},
 	}
 
 	h.handler = h.innerHandler(nil)
 	return h
 }
 
+func (h *handler) getCollector(project string, filters map[string]bool) (*collectors.MonitoringCollector, error) {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+
+	collectorKey := fmt.Sprintf("%s-%v", project, filters)
+	if c, ok := h.collectors[collectorKey]; ok {
+		return c, nil
+	}
+
+	collector, err := collectors.NewMonitoringCollector(project, h.m, collectors.MonitoringCollectorOptions{
+		MetricTypePrefixes:        h.filterMetricTypePrefixes(filters),
```

Comment:
I think we'll want to use this filtered value for the cache key. This way what's cached is what the collector is running.